mimic/rest/maas_api.py

```diff
@@ -477,23 +477,21 @@ def get_check(self, request, tenant_id, entity_id, check_id):
                methods=['PUT'])
     def update_check(self, request, tenant_id, entity_id, check_id):
         """
-        Update an existing check
+        Updates a check in place.
         """
-        checks = self._entity_cache_for_tenant(tenant_id).checks_list
-        newcheck = json.loads(request.content.read())
-        newcheck['entity_id'] = entity_id
-        for k in newcheck.keys():
```

Comment:
Encoding things on the way in is kind of an anti-pattern - see #320. As much as possible, we should keep objects however we got them and encode on the way out to APIs that depend on bytes being encoded a certain way.